Flaky test: TestContainerLogWithErrClosed

I noticed this test is randomly failing. We'd need to further investigate why it fails and in which conditions. It may be written in a way that is not consistent, or maybe not up-to-date. Opening this issue for awareness.

```
=== RUN   TestContainerLogWithErrClosed
2022/02/28 16:27:00 Starting container id: 07de0af25f64 image: docker:dind
2022/02/28 16:27:00 Waiting for container id 07de0af25f64 image: docker:dind
2022/02/28 16:27:17 Container is ready id: 07de0af25f64 image: docker:dind
    logconsumer_test.go:210: port not found
--- FAIL: TestContainerLogWithErrClosed (20.54s)
```

--- a/logconsumer_test.go
+++ b/logconsumer_test.go
@@ -2,6 +2,7 @@ package testcontainers
 
 import (
 	"context"
+	"errors"
 	"fmt"
 	"net/http"
 	"testing"
@@ -201,13 +202,29 @@ func TestContainerLogWithErrClosed(t *testing.T) {
 		},
 	})
 	if err != nil {
-		t.Fatal(err)
+		t.Fatal("create generic container:", err)
 	}
 	defer dind.Terminate(ctx)
 
-	remoteDocker, err := dind.Endpoint(ctx, "2375/tcp")
+	var remoteDocker string
+
+	ctxEndpoint, cancel := context.WithTimeout(ctx, 5*time.Second)
+	defer cancel()
+
+	// todo: remove this temporary fix (test is flaky).
+	for {
+		remoteDocker, err = dind.Endpoint(ctxEndpoint, "2375/tcp")
+		if err == nil {
+			break
+		}
+		if errors.Is(err, context.DeadlineExceeded) {
+			break
+		}
+		time.Sleep(100 * time.Microsecond)
+		t.Log("retrying get endpoint")
+	}
 	if err != nil {
-		t.Fatal(err)
+		t.Fatal("get endpoint:", err)
 	}
 
 	client, err := client.NewClientWithOpts(client.WithHost(remoteDocker))

